lib/core/pktmbuf/pktmbuf.c

```diff
@@ -105,37 +108,46 @@ pktmbuf_iterate(pktmbuf_info_t *pi, pktmbuf_cb_t cb, void *ud)
 }
 
 pktmbuf_info_t *
-pktmbuf_pool_create(char *addr, uint32_t bufcnt, uint32_t bufsz, uint32_t cache_sz, mbuf_ops_t *ops)
+pktmbuf_pool_cfg_create(const pktmbuf_pool_cfg_t *cfg)
 {
+    pktmbuf_pool_cfg_t _cfg, *c = &_cfg;
     pktmbuf_info_t *pi = NULL;
 
-    if (bufcnt == 0 || bufsz == 0)
-        CNE_ERR_GOTO(leave, "buffer count %d or buffer size %d is zero\n", bufcnt, bufsz);
+    if (!cfg)
+        goto leave;
+
+    /* Grab a copy of the const config data */
+    memcpy(c, cfg, sizeof(pktmbuf_pool_cfg_t));
 
-    CNE_MAX_SET(cache_sz, MEMPOOL_CACHE_MAX_SIZE);
+    if (c->bufcnt == 0 || c->bufsz == 0)
```

Comment:
This contradicts the code that occurs a few lines down which allows these to be set to 0 to indicate a default could be used. I'd prefer not to change those semantics.

```
CNE_DEFAULT_SET(c->bufcnt, 0, DEFAULT_MBUF_COUNT);
```

Reply:
restored to previous.